crates/dojo-lang/src/component.rs

```diff
@@ -139,6 +139,16 @@ pub fn handle_component_struct(
             .collect(),
     });
 
+    let member_prints: Vec<_> = members
+        .iter()
+        .map(|member| {
+            let member_name = &member.0;
+            format!("'{}'.print(); self.{}.print();", member_name, member_name)
+        })
+        .collect();
+
+    let print_body = member_prints.join("\n");
+
```

Comment:
Entrypoint. We construct the body of the implementation, printing the name and value